syntaxes/odoc.json

```diff
@@ -0,0 +1,169 @@
+  "fileTypes": ["mld"],
+  "patterns": [
+    {
+      "name": "comment.other.odoc",
+      "begin": "",
+      "end": "",
+      "patterns": [{ "include": "#markup" }]
+    },
+    {
+      "comment": "odoc documentation tag",
+      "name": "keyword.doc-tag.odoc",
+      "match": "\\@[[:lower:]]+"
+    }
+  ],
+  "repository": {
+    "markup": {
+      "patterns": [
+        {
+          "comment": "embedded ocaml source",
+          "begin": "(?<!\\\\)\\[",
+          "end": "(?<!\\\\)\\]",
+          "contentName": "source.embedded.odoc",
+          "patterns": [{ "include": "source.ocaml" }]
+        },
+        {
+          "comment": "label",
+          "begin": "\\{([[:digit:]]+)\\:([[:space:]]?)label\\b",
+          "end": "\\}",
+          "beginCaptures": {
+            "1": { "name": "variable.other.declaration.odoc" }
+          },
+          "contentName": "variable.other.declaration.odoc",
+          "patterns": [{ "include": "#markup" }]
+        },
+        {
+          "comment": "section header",
+          "begin": "\\{([[:digit:]]+)\\b",
+          "end": "\\}",
+          "beginCaptures": {
+            "1": { "name": "constant.language.odoc" }
+          },
+          "contentName": "constant.language.odoc",
+          "patterns": [{ "include": "#markup" }]
+        },
+        {
+          "comment": "text in bold",
+          "begin": "\\{b\\b",
+          "end": "\\}",
+          "contentName": "strong",
+          "patterns": [{ "include": "#markup" }]
+        },
+        {
+          "comment": "emphasize text",
+          "begin": "\\{e\\b",
+          "end": "\\}",
+          "contentName": "emphasis",
+          "patterns": [{ "include": "#markup" }]
+        },
+        {
+          "comment": "text in italic",
+          "begin": "\\{i\\b",
+          "end": "\\}",
+          "contentName": "emphasis",
+          "patterns": [{ "include": "#markup" }]
+        },
+        {
+          "comment": "center text",
+          "begin": "\\{C\\b",
+          "end": "\\}",
+          "patterns": [{ "include": "#markup" }]
+        },
+        {
+          "comment": "left align text",
+          "begin": "\\{L\\b",
+          "end": "\\}",
+          "patterns": [{ "include": "#markup" }]
+        },
+        {
+          "comment": "right align text",
+          "begin": "\\{R\\b",
+          "end": "\\}",
+          "patterns": [{ "include": "#markup" }]
+        },
+        {
+          "comment": "unordered list",
+          "begin": "\\{ul\\b",
+          "end": "\\}",
+          "patterns": [{ "include": "#markup" }]
+        },
+        {
+          "comment": "ordered list",
+          "begin": "\\{ol\\b",
+          "end": "\\}",
+          "patterns": [{ "include": "#markup" }]
+        },
+        {
+          "comment": "link address text",
+          "begin": "\\{\\{\\:\\b",
+          "end": "\\}",
+          "contentName": "string.other.odoc",
+          "patterns": [{ "include": "#markup" }]
```

Comment:
The second left curly bracket is unmatched and the text for the link is not highlighted:

![image](https://user-images.githubusercontent.com/25037249/140806197-4ee5bcd1-3708-4ae6-bbda-4b5cf037c7a1.png)

I think a combination of [regex lookahead](http://www.rexegg.com/regex-lookarounds.html) for the `{:` and adding a rule in the `patterns` will work

Reply:
Hi @mnxn, I've made the requested changes in your review but I have a bit of issue matching the link address text. Here's the regex I wrote for it, its still matches the string but doesn't match the link text.
<img width="631" alt="Screenshot 2021-11-12 at 11 17 27" src="https://user-images.githubusercontent.com/75541866/141450856-90057cac-898b-48b9-9053-b5c5786cd457.png">

Reply:
I was thinking something more like this

```json
"begin": "\\{(?=\\{\\:)",
"end": "\\}",
```

Then you can add a pattern to `patterns`:
```json
"patterns": [
    {
        "begin": "\\{\\:",
        "end": "\\}"
    },
    { "include":" #markup" }
]
```


I'll leave the highlighting names to you.

This way the parent pattern will match the outer `{` and `}` without matching the inner `{:`. The inner `{:` and `}` can be matched by the inner pattern.